src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy

```diff
@@ -1,358 +1,649 @@
+                validationResponse.verifyRuleSectionsAreEmpty(swId, ["missing", "excess", "misconfigured"])
+                validationResponse.verifyMeterSectionsAreEmpty(swId, ["missing", "excess", "misconfigured"])
             }
         }
 
-        and: "Involved switches pass switch validation"
-        [isl.srcSwitch, isl.dstSwitch].each { sw ->
-            verifyAll(northbound.validateSwitch(sw.dpId)) {
-                rules.missing.empty
-                rules.excess.empty
-                rules.misconfigured.empty
-                meters.missing.empty
-                meters.excess.empty
-                meters.misconfigured.empty
+        and: "Expect ISL RTT for new ISL in forward/reverse directions"
+        wait(islSyncWaitSeconds, 2) {
+            checkIslRttStats(newIsl, checkpointTime, true)
+            checkIslRttStats(newIsl.reversed, checkpointTime, true)
+        }
+
+        and: "Expect no ISL RTT for MOVED ISL in forward/reverse directions"
+        checkIslRttStats(isl, checkpointTime, false)
+        checkIslRttStats(isl.reversed, checkpointTime, false)
+
+        when: "Replug the link back where it was"
+        islUtils.replug(newIsl, true, isl, false, false)
+        islUtils.waitForIslStatus([isl, isl.reversed], DISCOVERED)
+        def originIslIsUp = true
+        islUtils.waitForIslStatus([newIsl, newIsl.reversed], MOVED)
+
+        and: "Remove the MOVED ISL"
+        assert northbound.deleteLink(islUtils.toLinkParameters(newIsl)).size() == 2
+        def newIslIsRemoved = true
+
+        then: "Server42 ISL RTT rules are deleted on the dst switch of the moved link"
+        wait(RULES_DELETION_TIME) { checkIslRttRules(newIsl.dstSwitch.dpId, true) }
+
+        and: "All involved switches pass switch validation"
+        [isl.srcSwitch.dpId, isl.dstSwitch.dpId, newIsl.dstSwitch.dpId].each { swId ->
+            with(northbound.validateSwitch(swId)) { validationResponse ->
+                validationResponse.verifyRuleSectionsAreEmpty(swId, ["missing", "excess", "misconfigured"])
+                validationResponse.verifyMeterSectionsAreEmpty(swId, ["missing", "excess", "misconfigured"])
+            }
+        }
+
+        cleanup:
+        revertToOrigin(islRttFeatureStartState, initialSwitchRtt)
+        if (!originIslIsUp) {
+            islUtils.replug(newIsl, true, isl, false, true)
+            islUtils.waitForIslStatus([isl, isl.reversed], DISCOVERED)
+        }
+        if (newIsl && !newIslIsRemoved) {
+            northbound.deleteLink(islUtils.toLinkParameters(newIsl))
+            wait(WAIT_OFFSET) {
+                assert !islUtils.getIslInfo(newIsl).isPresent()
+                assert !islUtils.getIslInfo(newIsl.reversed).isPresent()
             }
         }
+        database.resetCosts()
+    }
+
+    @Tidy
+    @Tags([HARDWARE])
+    def "ISL RTT stats are available for one direction in case link is UP in forward direction only"() {
```

Comment:
Can you review this test? It seems that test name does not correlate with the actual behavior